management-microservice/src/main/java/com/nearsoft/fgaribay/mgmt/microservice/ProductManagementService.java

```diff
@@ -0,0 +1,46 @@
+package com.nearsoft.fgaribay.mgmt.microservice;
+
+import com.nearsoft.fgaribay.mgmt.ProductRepository;
+import com.nearsoft.fgaribay.mgmt.model.Product;
+import com.nearsoft.fgaribay.mgmt.model.ProductRequest;
+import com.nearsoft.fgaribay.mgmt.model.ProductResponse;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.UUID;
+
+@Component
+public class ProductManagementService {
+
+  private ProductRepository productRepository;
+
+  @Autowired
+  public ProductManagementService(ProductRepository productRepository) {
+    this.productRepository = productRepository;
+  }
+
+  @RabbitListener(queues = "${product.queues.list-name}")
+  public ProductResponse listProducts(ProductRequest request) {
+
+    ArrayList<Product> products = new ArrayList<>(productRepository.getAllProducts());
```

Comment:
Using `List` you avoid to create a new array list with the same results you get from the repository. With that, the repository can define which `List` implementation uses to give you the data and the objects using the result won't care at all of the final type since it will be always a `List`.